src/main/java/org/iitbact/cc/services/FacilityServices.java

```diff
@@ -63,12 +57,10 @@ public Facility createFacility(FacilityRequest request) throws CovidControlExcep
 		log.info("Create Facility Request");
 		log.debug("Create Facility Request - {}", facility.toString());
 
-		facilityRepository.save(facility);
-
-		FacilityContact contact = FacilityContact.builder().facility(facility).facilityId(facility.getFacilityId())
-				.data(request.getContactDetails().generateJson(jsonFactory)).build();
-		facility.setFacilityContact(contact);
-
+		if(null != facility.getFacilityId()){
+			throw new CovidControlException(new CovidControlErpError(CovidControlErrorCode.FACILITY_ID_SHOULD_BE_NULL, CovidControlErrorMsg.FACILITY_ID_SHOULD_BE_NULL));
+		}
+		facility.getFacilityContact().setFacility(facility);
```

Comment:
Since we are directly asking it as part of facilityContact, we don't need to create another object